### PR TITLE
RD2: Display data migration errors

### PR DESCRIPTION
**Pull request by RadSFDeveloper**
Display a summary of the records processed during RD migration to indicate what was completed and if there were issues, what needs to be resolved.

# Critical Changes

# Changes

# Issues Closed

# New Metadata
Lightning Component Bundle:
- util

Custom Field:
- npe03__Recurring_Donations_Settings__c.RecurringDonations2MigrationState__c

Custom Labels:
- BatchProgressStatusComplete
- BatchProgressStatusCompleteErrors
- BatchProgressStatusError
- BatchProgressStatusFailed
- BatchProgressStatusHolding
- BatchProgressStatusPreparing
- BatchProgressStatusProcessing
- BatchProgressStatusQueued
- BatchProgressStatusStopped
- BatchProgressStatusSuccess
- BatchProgressTotalRecords
- BatchProgressTotalRecordsFailed
- BatchProgressTotalRecordsProcessed
- RD2_EnablementMigrationErrorMessage

# Deleted Metadata

### CODE REVIEW

**Comment by RadSFDeveloper** [comment]
**lurch: attach W-037818

**Comment by LurchTheButler** [comment]
Tracking <a href="https://foundation.my.salesforce.com/a2x1E000001HpIvQAK" target="_blank">W-037818</a>

File: src/objects/npe03__Recurring_Donations_Settings__c.object [hunk 1]
```diff
@@ -14,6 +14,18 @@
         <type>Number</type>
         <unique>false</unique>
     </fields>
+    <fields>
+        <fullName>RecurringDonations2MigrationState__c</fullName>
```
Comment (RadSFDeveloper) [inline src/objects/npe03__Recurring_Donations_Settings__c.object]: @sam-knox , can you please review the field containing dry run and data migration results?

File: src/labels/CustomLabels.labels [hunk 2]
```diff
@@ -1564,7 +1668,15 @@ If you&apos;re not sure how to resolve these errors, post a message in the Nonpr
         <language>en_US</language>
         <protected>true</protected>
         <shortDescription>RD2_EnablementMigrationCompleteMessage</shortDescription>
-        <value>&lt;b&gt;&lt;i&gt;All done! You’re now running the Enhanced Recurring Donations experience!&lt;/i&gt;&lt;/b&gt;</value>
+        <value>Your migration was successful with no errors, you’re all ready to go with Enhanced Recurring Donations.</value>
+    </labels>
+    <labels>
+        <fullName>RD2_EnablementMigrationErrorMessage</fullName>
```
Comment (RadSFDeveloper) [inline src/labels/CustomLabels.labels]: @sam-knox , can you please review data migration complete message displayed when data migration completes successfully and the data migration error message displayed when data migration completes with errors?

File: src/objects/npe03__Recurring_Donations_Settings__c.object (hunk 1 above)
Comment (sam-knox) [reply src/objects/npe03__Recurring_Donations_Settings__c.object]: @RadSFDeveloper - Would this label only be visible in the settings object, or also on the RD2 enablement page in NPSP Settings?

Comment (RadSFDeveloper) [reply src/objects/npe03__Recurring_Donations_Settings__c.object]: Only on the RD settings.

File: src/labels/CustomLabels.labels (hunk 2 above)
Comment (sam-knox) [reply src/labels/CustomLabels.labels]: @RadSFDeveloper How about this:

"Your migration was successful with no errors. All set up steps are now complete. You're ready to begin using Enhanced Recurring Donations."

Comment (RadSFDeveloper) [reply src/labels/CustomLabels.labels]: It has been done.

File: src/objects/npe03__Recurring_Donations_Settings__c.object (hunk 1 above)
Comment (sam-knox) [reply src/objects/npe03__Recurring_Donations_Settings__c.object]: @RadSFDeveloper - Cool, then I think what you have here works fine.

**Comment by RadSFDeveloper** [comment]
@Julian88Tex , data migration error display is ready for testing. Thank you!

**Pull request merged**